src/ctapipe/irf/optimize.py

```diff
@@ -0,0 +1,483 @@
+"""module containing optimization related functions and classes"""
+
+import operator
+from abc import abstractmethod
+
+import astropy.units as u
+import numpy as np
+from astropy.io import fits
+from astropy.table import QTable, Table
+from pyirf.cut_optimization import optimize_gh_cut
+from pyirf.cuts import calculate_percentile_cut, evaluate_binned_cut
+
+from ..core import Component, QualityQuery
+from ..core.traits import AstroQuantity, Float, Integer
+from .binning import ResultValidRange, make_bins_per_decade
+from .select import EventPreProcessor
+
+
+class OptimizationResult:
+    def __init__(self, precuts, valid_energy, valid_offset, gh, theta):
+        self.precuts = precuts
+        self.valid_energy = ResultValidRange(valid_energy, "energy")
+        self.valid_offset = ResultValidRange(valid_offset, "offset")
+        self.gh_cuts = gh
+        self.theta_cuts = theta
+
+    def __repr__(self):
+        if self.theta_cuts is not None:
+            return (
+                f"<OptimizationResult with {len(self.gh_cuts)} G/H bins "
+                f"and {len(self.theta_cuts)} theta bins valid "
+                f"between {self.valid_offset.min} to {self.valid_offset.max} "
+                f"and {self.valid_energy.min} to {self.valid_energy.max} "
+                f"with {len(self.precuts.quality_criteria)} precuts>"
+            )
+        else:
+            return (
+                f"<OptimizationResult with {len(self.gh_cuts)} G/H bins valid "
+                f"between {self.valid_offset.min} to {self.valid_offset.max} "
+                f"and {self.valid_energy.min} to {self.valid_energy.max} "
+                f"with {len(self.precuts.quality_criteria)} precuts>"
+            )
+
+
+class OptimizationResultStore:
```

Comment:
Why two classes for the data itself and methods to just read and write them?

Why not add the io related methods to the class itself or have simple, free functions to read / store it

Reply:
If I remember correctly, the idea was to separate the definition of the output product of the optimization tool from the code generating it.
@Tobychev Do you think it is worth keeping it this way?